control/server.py

```diff
@@ -163,9 +181,11 @@ def _start_spdk(self):
             cmd += shlex.split(spdk_tgt_cmd_extra_args)
         self.logger.info(f"Starting {' '.join(cmd)}")
         try:
-            self.spdk_process = subprocess.Popen(cmd,
-                                                 preexec_fn=set_pdeathsig(
-                                                     signal.SIGTERM))
+            # install SIGCHLD handler
+            signal.signal(signal.SIGCHLD, sigchld_handler)
+
+            # start spdk process
+            self.spdk_process = subprocess.Popen(cmd)
```

Comment:
Did you consider using an [existing daemon implementation](https://peps.python.org/pep-3143/#reference-implementation)?

Reply:
Thank you for the idea, @epuertat , I will 👀 into it.